docs/guide/topics/permissions.rst

```diff
@@ -0,0 +1,222 @@
+
+    # So we want to create a new role, and to give that role some permissions.
+    # We will define a set of `Permissions` that do what we need:
+    permissions = discord.Permissions()
+    permissions.send_messages = True
+    permissions.add_reactions = True
+    permissions.create_public_threads = True
+
+    # Above is now a working Permissions instance with the name `permissions`, we could also define it like so:
+    permissions = discord.Permissions(send_messages=True, add_reactions=True, create_public_threads=True)
+
+    # and we pass this to the role creation method!
+    await guild.create_role(name="My cool game", mentionable=True, permissions=permissions, reason="New game!")
+
+Now we have a brand new role, named ``"My cool game"`` and it now also has the permissions we outlined above.
+
+Setting permissions on an existing channel
+-------------------------------------------
+
+Let's say you wanted to edit the permissions of an existing channel, perhaps to give a role more access.
+An example here is that my new moderator role needs the Manage Messages permission in the #general channel.
+We will be using the :meth:`~TextChannel.set_permissions` method:
+
+.. code-block:: python3
+
+    # Let's create a PermissionOverwrite instance that has some explicit permissions set:
+    overwrite = discord.PermissionOverwrite()
+    overwrite.update(send_messages=True, send_messages_in_threads=True)
+
+    # and now we update that with `manage_messages`:
+    overwrite.manage_messages = True
+
+    # we can now set those permissions on the channel:
+    await channel.set_permissions(moderator_role, overwrite=overwrite)
+
+This short example briefly goes over PermissionOverwrite objects, let's look more into those now:
+
+Using ``PermissionOverwrite`` objects during channel creation
+--------------------------------------------------------------
+
+We can also create channels with permissions set from the start.
+Let's create a text channel where only the moderator role, and someone trying to make a report, can see the channel.
+
+We need to construct a mapping of ``Role | Member: PermissionOverwrite``:
+
+.. code-block:: python3
+
+    # We can now create the overwrites mapping necessary for this, you can do it in one dictionary instantiation, or spead it across multiple lines.
+
+    overwrites = {
+        guild.default_role: discord.PermissionOverwrite(view_channel=False, send_messages=False, read_messages=False, read_message_history=False),
+        moderator_role: discord.PermissionOverwrite(manage_messages=True),
+        member: discord.PermissionOverwrite(view_channel=True, send_messages=True, read_messages=True, read_message_history=True)
+    }
+
+    # or alternatively:
+    overwrites = {}
+    overwrites[guild.default_role] = discord.PermissionOverwrite(view_channel=False, send_messages=False, read_messages=False, read_message_history=False)
+    overwrites[moderator_role] = discord.PermissionOverwrite(manage_messages=True)
```

Comment:
```suggestion
    overwrites[moderator_role] = discord.PermissionOverwrite(view_channel=True, send_messages=True, read_messages=True, read_message_history=True, manage_messages=True)
```
nit-picky here but moderator needs to be able to view channel and send/read messages